### ISSUE TITLE
Squashing commits fails silently

### ISSUE THREAD

**Issue by danfrankj** [airbnb/knowledge-repo#236]
`knowledge_repo add --update  --squash ~/path/to.ipynb -p my/path.kp`

I think we need a forced git push here

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj

**Comment by matthewwardrop** [comment]
Yeah... my bad. I've been aware of this for a while, but put it off since I want to rewrite the Git backend a bit. I'll look into this when I can :).
